gohypergiant/hyperdrive
# MLSDK-414: FastApp | Handle AutoML/PyTorch model outputs

## Summary

_A brief summary of the objective._
Correctly process AutoML results, in order to get sensible predictions.

_Describe your changes._
Changed the `fastapp` predict endpoint/function so that, based on the ML task (e.g., binary classification, multi-class classification, regression), the predictions returned from the AutoML model are now sensible. The fastapp predictions are now processed in the same way that they are in the [research-EfficientAutoML](https://github.com/gohypergiant/research-EfficientAutoML) library.

## Testing

_Describe steps to follow to verify the changes that were made._
1. Make sure the `hyper` CLI is installed. If not installed, see this link: [Hyper CLI Installation](https://github.com/gohypergiant/hyperdrive/blob/stable/hyper/README.md#installing-from-source).
2. Make sure `hyperpackage` is installed. If not installed, see this link: [hyperpackage installation](https://github.com/gohypergiant/hyperdrive/blob/stable/hyper/README.md#installing-from-source).
3. Git clone the [research-EfficientAutoML](https://github.com/gohypergiant/research-EfficientAutoML) repo if you don't already have it on your machine.
4. Navigate to the `research-EfficientAutoML` repo in a terminal session.
5. Run the [AutoML binary classification model starter notebook](https://github.com/gohypergiant/research-EfficientAutoML/blob/development/starter_notebooks/binary_classification_starter.ipynb). You'll need to get the `output` object.
6. Run this in the previously referenced starter notebook: `from hyperpackage.hyperpack_creation import create_hyperpack`.
7. Then run this in the next cell of the notebook: `create_hyperpack(trained_model=output["model"], model_flavor="automl", ml_task="binary_classification")`.
8. In a terminal session, navigate to the `research-EfficientAutoML` repo.
9. Run `hyper pack`.
10. Obtain a prediction by following the Cookbook instructions, specifically the section entitled ["Submit a prediction"](https://github.com/gohypergiant/research-EfficientAutoML).
11. Since this is binary classification model, the prediction should be either a `0` or `1`.

_If this ticket is a low-fi ticket specify what if anything is testable. A
"not testable" is perfectly acceptable in this case._
N/A

_Point out any previously tested functionality that may have been changed
and need another look._
N/A

## Notes

_Optional_

_Extra details about the PR that don't fit elsewhere such as dependencies
on another branch, a known issue that needs extra attention, or links._

## Screenshots

_Optional_

## Ticket(s)

- [MLSDK-414](https://gohypergiant.atlassian.net/browse/MLSDK-414)
